endpoint.go

```diff
@@ -93,6 +94,7 @@ func (ep *endpoint) MarshalJSON() ([]byte, error) {
 	epMap["myAliases"] = ep.myAliases
 	epMap["svcName"] = ep.svcName
 	epMap["svcID"] = ep.svcID
+	epMap["virtualIP"] = ep.virtualIP.String()
```

Comment:
Shouldn't we add it to the map only if present, if not nil ?
IP.String() may handle a nil IP, but I am not sure how the unmarshal is going to behave.